tests/functionality_tests/mission_test.py

```diff
@@ -0,0 +1,54 @@
+from engine.robot import Robot
+from engine.base_station import BaseStation
+from engine.mission import Mission
+from engine.mission import ControlMode
+from engine.database import DataBase
+from engine.robot_state import Robot_State
+from engine.phase import Phase
+
+import math
+
+
+class MissionTest():
+
+    def __init__(self):
+        self.rb_state = Robot_State(0, 0, 0, .2, .5, .2)
+        self.rob = Robot(self.rb_state)
+        self.rob_base = BaseStation(position=(
+            42.444250, -76.483682), heading=math.pi/2, battery=None, plastic_load=None)
+        self.database = DataBase(self.rob)
+        self.m = Mission(robot=self.rob, base_station=self.rob_base,
+                         init_control_mode=ControlMode.LAWNMOWER)
+
+    def test_setup(self):
```

Comment:
can we add specifications for all of these test functions? just by looking at it i dont really know what it's testing per se, like it's calling the execute_setup function but are we actually checking anything? why do we stop the obstacle avoidance thread?

Reply:
got it got it - lets add this at the top of the file in a comment just to clarify it a little bit!

Reply:
would it be better to comment where I disabled the thread?

Reply:
ill just put it in both places